raiden/ui/web/src/app/components/channel-table/channel-table.component.ts

```diff
@@ -26,37 +25,43 @@ export class ChannelTableComponent implements OnInit {
     public tempChannel: Channel;
     public watchEvents: EventsParam[] = [];
     public tabIndex = 0;
+    private channelsSubject: BehaviorSubject<void> = new BehaviorSubject(null);
 
     constructor(
         private raidenConfig: RaidenConfig,
         private raidenService: RaidenService,
         private sharedService: SharedService,
-    ) { }
+    ) {
+    }
 
     ngOnInit() {
         let timeout;
         this.channels$ = this.channelsSubject.pipe(
             tap(() => clearTimeout(timeout)),
             switchMap(() => this.raidenService.getChannels()),
-            map((newChannels) =>
-                newChannels.map((newchannel) =>
-                    Object.assign(newchannel, { menu: null }) as WithMenu<Channel>
-                )
-            ),
+            map((newChannels) => newChannels.map((newchannel) =>
+            Object.assign(newchannel, {menu: null}) as WithMenu<Channel>
+        )),
             scan((oldChannels, newChannels) => {
                 // use scan and Object.assign to keep object references and
                 // improve *ngFor change detection on data table
                 for (const newchannel of newChannels) {
                     const oldchannel: WithMenu<Channel> = oldChannels.find((c) =>
                         c.channel_identifier === newchannel.channel_identifier);
-                    if (oldchannel && oldchannel.state == newchannel.state) {
-                        Object.assign(newchannel, { menu: oldchannel.menu });
+
+                    if (oldchannel && oldchannel.state === newchannel.state) {
```

Comment:
Can you please also add `balance` comparison here to trigger `updateMenu`, as this is also a condition that may change item availability?

Reply:
I changed the `updateMenu` to trigger either way so the balance will be checked inside the update menu.